agent/consul/leader_connect_ca.go

```diff
@@ -1451,6 +1451,21 @@ func (c *CAManager) SignCertificate(csr *x509.CertificateRequest, spiffeID conne
 		defer c.caLeafLimiter.csrConcurrencyLimiter.Release()
 	}
 
+	inter, err := provider.ActiveIntermediate()
+	if err != nil {
+		return nil, err
+	}
+	root, err := provider.ActiveRoot()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I forget if we discussed this already, but do we need to do this `root` lookup here? It is only used later on to compare them. Since we aren't checking the root expiration maybe this can stack where it was?

Reply:
Actually you bring a good point, I'm not sure about that and may be we need input from other folks but root expiration need to be checked as well. My understanding is that any cert in the cert chain that expire invalidate the whole chain.

Reply:
After discussion with @banks he confirmed my thoughts. I added the root expiry check.